File: PipelineService/Models/MqttMessages/BlockExecutionResponse.cs
using System;

namespace PipelineService.Models.MqttMessages
{
    public abstract class BlockExecutionResponse : BaseMqttMessage
    {
        /// <summary>
        /// The pipeline's id the block was executed for.
        /// </summary>
        public Guid PipelineId { get; set; }

        /// <summary>
        /// The pipeline execution this block belongs to. 
        /// </summary>
        public Guid ExecutionId { get; set; }

        /// <summary>
        /// The block's id that was executed.
        /// </summary>
        public Guid BlockId { get; set; }

        /// <summary>
        /// Indicates if the execution was successful.
        /// </summary>
        public bool Successful { get; set; }

        /// <summary>
        /// The time (UTC) the execution of this block was started.  
        /// </summary>
        public DateTime StartTime { get; set; }

        /// <summary>
        /// The time (UTC) the execution of this block was stopped (either due to an error or completion).
        /// </summary>
        public DateTime StopTime { get; set; }
    }
}

File: BlockWorker/src/models/block_execution_response.py
import datetime
import json

from src.models.message import Message


class BlockExecutionResponse(Message):
    pipeline_id: str
    execution_id: str
    block_id: str
    successful: bool
    start_time: datetime
    stop_time: datetime

    def __init__(self):
        super().__init__()

        self.start_time = datetime.datetime.now(datetime.timezone.utc)

    def set_pipeline_id(self, pipeline_id: str):
        self.pipeline_id = pipeline_id

    def set_block_id(self, block_id: str):
        self.block_id = block_id

    def set_execution_id(self, execution_id: str):
        self.execution_id = execution_id

    def set_successful(self, successful: bool):
        self.successful = successful

    def set_start_time(self, start_time: datetime):
        self.start_time = start_time

    def set_stop_time(self, stop_time: datetime):
        self.stop_time = stop_time
